sources/web/datalab/polymer/notebook.ts

```diff
@@ -83,6 +84,25 @@ async function processMessageEvent(e: MessageEvent) {
       sendMessageToNotebookEditor(outgoingMessage);
       break;
 
+    case CommandId.REDIRECT_TO_NOTEBOOK:
+      try {
+        const id = message.arguments;
+        const url = '/notebook/' + id;
+        window.location.href = url;
+        outgoingMessage = {
```

Comment:
Does this work reliably? Doesn't the whole app stop executing after the browser switches away? Perhaps we should send the async reply message then set `location.href`?

Reply:
It seems to take a while for the setting of href to take effect. I initially had a bug in some of the code after the href line, and I was getting an error dialog that blinked up before it switched to the other location. I assumed we needed not to have anything important running after setting href, so as long as it doesn't do anything bad, I'm not worried about whether it actually executes anything past that point.

Reply:
I'm more concerned about that code potentially not getting executed at all. I think running any code after setting `location.href` is at least unpredictable.